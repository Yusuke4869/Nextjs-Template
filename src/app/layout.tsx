import { Noto_Sans_JP } from "next/font/google";
import "~/styles/globals.css";

import type { FC, PropsWithChildren } from "react";

const notoSansJp = Noto_Sans_JP({
  subsets: ["latin"],
  variable: "--font-noto-sans-jp",
});

const RootLayout: FC<PropsWithChildren> = ({ children }) => (
  <html lang="ja">
    <body className={notoSansJp.className}>
      <main>{children}</main>
    </body>
  </html>
);

export default RootLayout;
