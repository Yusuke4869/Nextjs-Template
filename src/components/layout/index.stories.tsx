import { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import Layout from ".";

type T = typeof Layout;
type Story = ComponentStoryObj<T>;

const data = {
  pageTitle: "Hello World",
  pageDescription: "ページ説明",
  pageImage: "https://example.com",
  children: "content",
};

export default {
  component: Layout,
  args: { ...data },
  argTypes: {
    pageTitle: {
      description: "ページタイトル",
      control: {
        type: "text",
      },
    },
    pageDescription: {
      description: "ページ説明",
      control: {
        type: "text",
      },
    },
    pageImage: {
      description: "OGP Image URL",
      control: {
        type: "text",
      },
    },
    children: {
      description: "子要素",
      control: {
        type: "text",
      },
    },
  },
} as ComponentMeta<T>;

export const Default: Story = {};
