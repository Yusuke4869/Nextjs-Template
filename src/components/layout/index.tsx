import Head from "next/head";
import type { FC, ReactNode } from "react";
import Footer from "~/components/layout/footer";
import Header from "~/components/layout/header";

interface Information {
  pageTitle: string;
  pageDescription: string;
  pageImage: string;
}

type Props = {
  children: ReactNode;
  pageTitle: string;
  pageDescription?: string;
  pageImage?: string;
};

const DefaultDescription = "";
const DefaultImage = "";

const Metas: FC = () => (
  <>
    <meta charSet="utf-8" />
    <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    <meta name="theme-color" content="#ffffff" />
  </>
);

const SeoMetas: FC<Information> = ({ pageTitle, pageDescription, pageImage }) => (
  <>
    <meta name="name" content={pageTitle} />
    <meta name="image" content={pageImage} />
    <meta name="description" content={pageDescription} />
  </>
);

const OgpMetas: FC<Information> = ({ pageTitle, pageDescription, pageImage }) => (
  <>
    <meta property="og:title" content={pageTitle} />
    <meta property="og:description" content={pageDescription} />
    <meta property="og:image" content={pageImage} />
    <meta property="og:image:alt" content="" />
    <meta property="og:site_name" content="" />
  </>
);

const TwitterMetas: FC<Information> = ({ pageTitle, pageDescription, pageImage }) => (
  <>
    <meta name="twitter:card" content="summary_large_image" />
    <meta name="twitter:site" content="" />
    <meta name="twitter:title" content={pageTitle} />
    <meta name="twitter:description" content={pageDescription} />
    <meta name="twitter:image" content={pageImage} />
  </>
);

const Layout: FC<Props> = ({ children, pageTitle, pageDescription = DefaultDescription, pageImage = DefaultImage }) => (
  <>
    <Head>
      <title>{pageTitle}</title>
      <Metas />
      <SeoMetas pageTitle={pageTitle} pageDescription={pageDescription} pageImage={pageImage} />
      <OgpMetas pageTitle={pageTitle} pageDescription={pageDescription} pageImage={pageImage} />
      <TwitterMetas pageTitle={pageTitle} pageDescription={pageDescription} pageImage={pageImage} />
    </Head>
    <main>
      <div>
        <Header />
        {children}
      </div>
      <Footer />
    </main>
  </>
);

export default Layout;
