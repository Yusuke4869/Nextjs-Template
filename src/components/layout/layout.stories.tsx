import { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import Layout from ".";

type T = typeof Layout;
type Story = ComponentStoryObj<T>;

const data = {
  PageTitle: "Hello World",
  PageDescription: "ページ説明",
  PageImage: "https://example.com",
  children: "content",
};

export default {
  component: Layout,
  args: { ...data },
  argTypes: {
    PageTitle: {
      description: "ページタイトル",
      control: {
        type: "text",
      },
    },
    PageDescription: {
      description: "ページ説明",
      control: {
        type: "text",
      },
    },
    PageImage: {
      description: "OGP Image URL",
      control: {
        type: "text",
      },
    },
    children: {
      description: "子要素",
      control: {
        type: "text",
      },
    },
  },
} as ComponentMeta<T>;

export const Default: Story = {};
