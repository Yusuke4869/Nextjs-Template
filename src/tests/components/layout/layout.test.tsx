import { composeStories } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import * as stories from "~/components/layout/layout.stories";

const { Default } = composeStories(stories);

describe("(components) layout", () => {
  test("Snap Shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
