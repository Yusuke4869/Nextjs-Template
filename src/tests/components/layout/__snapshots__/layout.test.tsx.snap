// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components) layout Snap Shot 1`] = `
<div>
  <main>
    <div>
      <header>
        <p>
          Header
        </p>
      </header>
      content
    </div>
    <footer>
      <p>
        Footer
      </p>
    </footer>
  </main>
</div>
`;
